File: ui/react/src/action/RemoveRows.ts
import TableAction from "./TableAction";
import Action from "./Action";

class RemoveRows extends TableAction {
    public rowIndex: number;
    public nRows: number;

    public static deserialize(input: any) {
        const sheetName: string = input.sheetName;
        const tableName: string = input.tableName;
        const rowIndex: number = input.rowIndex;
        const nRows: number = input.nRows;
        return new RemoveRows(sheetName, tableName, rowIndex, nRows);
    }

    constructor(
        sheetName: string,
        tableName: string,
        rowIndex: number,
        nRows: number) {

        super(sheetName, tableName);
        this.rowIndex = rowIndex;
        this.nRows = nRows;
    }

    public wrapper(): Action {
        const action = new Action();
        action.removeRows = this;
        return action;
    }

}

export default RemoveRows;

File: core/src/main/java/com/ctrip/ferriswheel/core/action/RemoveRows.java
package com.ctrip.ferriswheel.core.action;

import com.ctrip.ferriswheel.api.action.Action;

public class RemoveRows extends TableAction implements Action {
    private int rowIndex;
    private int nRows;

    public RemoveRows() {
    }

    public RemoveRows(String sheetName, String tableName, int rowIndex, int nRows) {
        super(sheetName, tableName);
        this.rowIndex = rowIndex;
        this.nRows = nRows;
    }

    public int getRowIndex() {
        return rowIndex;
    }

    public void setRowIndex(int rowIndex) {
        this.rowIndex = rowIndex;
    }

    public int getnRows() {
        return nRows;
    }

    public void setnRows(int nRows) {
        this.nRows = nRows;
    }
}
